### ISSUE TITLE
Second pass at new SCORM-replacing content (MUST be done before 1.0.3)

### ISSUE THREAD

**Issue by garemoko** [adlnet/xAPI-Spec#628]
See: 
* https://github.com/adlnet/xAPI-Spec/issues/501
* https://github.com/adlnet/xAPI-Spec/pull/540
* https://github.com/adlnet/xAPI-Spec/pull/606#discussion_r24931270

@fugu13 has commented that he wants to do a closer review of the new content designed to remove our dependancy on SCORM before we release 1.0.3. I fully support this, especially because I'm fielding a number of questions from developers about these section (this is my main prompt for not wanting the master branch to be the development branch). 

Specifically, the following areas need attention:

* The [interaction types table](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#interaction-types)
* The [response patterns table](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#response-patterns)
* The [character string parameters section](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#characterstring-parameters) and associated [requirements](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#requirements-3)
* The description column of the [interaction components list table](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#interaction-components)
* The description column of the [result score table](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI.md#4151-score)

**Comment by earleskinner** [comment]
This might be totally irrelevant or the wrong place for it, but I started writing this (never got a chance to finish it) https://github.com/earleskinner/Cog/blob/master/src/Cog/Xapi/ActivityDefinitons/Readme.md. Hopefully it helps.

**Comment by garemoko** [comment]
@earleskinner Thanks for sharing! This particular issue is about removing direct dependancies on SCORM and pulling the relevant requirements into the spec as they are. There's no change in behaviour of either the LRS or activity provider. 

It looks like your document is proposing changes in behaviour (though I think you need to finish it to make clear what those changes are). So that would be a [new issue](https://github.com/adlnet/xAPI-Spec/issues/new).

**Comment by garemoko** [comment]
@fugu13 do you think this issue can now be closed as we've had a couple of PRs (#606 #649) in this area now and no new issues recently?

**Comment by fugu13** [comment]
I'll do another pass for full nonambiguous implementability soon, then we can ditch this

**Comment by falcon-git** [comment]
The responsePattern section in 1.0.3 will probably help a lot. Good work!

Maybe also make some of the examples in appendix C more complex so that it is more revealing? For instance make it include complete xAPI statements including the result with response, fill-in with multiple blanks and case sensitivity etc.

**Comment by garemoko** [comment]
+1 Not sure we need full statements, but more complex correctResponses and examples of corresponding results will certainly be beneficial.

**Comment by andyjohnson** [comment]
Probably closable after #764 and #747

**Comment by garemoko** [comment]
Making a separate issue to tackle the additional examples in section C suggested by @falcon-git

**Comment by falcon-git** [comment]
Where does the delimiters like [,] stem from? Scorm?

Is there an issue for moving away from this strange pattern formats and start using json instead?

i.e. "{case_matters=false}blue[,]Cloud[,]black"

Would probably be better described as:

{
  caseMatters: false,
  correct answers: [
    ['blue'], ['Cloud'], ['black']
  ]
}

Better because you wouldn't have to create custom code to parse the data, and it would be easier to add more properties later like support for multiple correct alternatives.

Or maybe there are good reasons for using the current format that I'm not aware of?

**Comment by garemoko** [comment]
@falcon-git short answer: Yes, it comes from SCORM. Yes, it would be good to move to a JSON format. No, we can't do it in this version due to compatibility with existing adopters. 

Longer answer: If we get to the point where we have enough benefits to adopters to justify breaking changes and a 2.0 version, then this is definitely one thing I'd like to change. That said, by that point Learning as an industry might have moved far enough from the SCORM model by that point that we don't need these properties at all in the core spec at all. I'm actually become less and less confident of this though as I see the applicability of these kinds of interactions in things like VR simulations where you have to make a choice between certain actions or objects.

**Comment by falcon-git** [comment]
What is the practice. Are people creating issues for 2.0 now? Where are the ideas stored?

**Comment by falcon-git** [comment]
And thanks a lot for a good answer :)

**Comment by garemoko** [comment]
Yes, issues can be created here and tagged "major".

**Comment by falcon-git** [comment]
Thanks, I've made an issue for this, but was unable to label it.
